src/Writer/XlsWriter.php

```diff
@@ -78,8 +78,11 @@ public function write(array $data): void
         $this->init($data);
 
         fwrite($this->file, '<tr>');
+        // prevent xls injection
+        $patterns = ['/^=/', '/^\+/', '/^-/', '/^@/'];
+        $replace = ['!=', '!+', '!-', '!@'];
         foreach ($data as $value) {
-            fwrite($this->file, sprintf('<td>%s</td>', $value));
+            fwrite($this->file, sprintf('<td>%s</td>', preg_replace($patterns, $replace, $value)));
```

Comment:
In case people actually rely on the values not being escaped, should their be an option to disable the escaping?

Reply:
Yes I was just thinking maybe this could be a configurable option. I had a client run a pen test on an application using Sonata Admin with the Exporter bundle and they identified this vulnerability - and I had to fix it or else they weren't going to use our application. I didn't like the idea of manipulating the cell data but it was the only option I could find to "plug the hole".

If this isn't useful for the community that's totally fine, for now I've just made my own CsvWriter and XlsWriter classes and use those in place of the ones in this bundle, but that makes me have to closely monitor the changes in those classes when I want to update this bundle - which is no fun at all...

Reply:
> In case people actually rely on the values not being escaped, should their be an option to disable the escaping?

I have now made it optional, defaulting to not used (**false**) so any existing applications using this bundle will not experience any change unless they explicitly add the **safe_cells** configuration parameter and set it to **true** for one or both writers.